### PR TITLE
Clarified ServiceVersion Guidelines

### PR DESCRIPTION
**Pull request by KrzysztofCwalina**
This guideline update clarifies what we have been doing already, i.e. we reserved the ServiceVersion value 0, mainly so people don't accidentally pass in the default value.

### CODE REVIEW

File: docs/dotnet/introduction.md [hunk 1]
```diff
@@ -681,7 +681,9 @@ For example, the following is a code snippet from the `ConfigurationClientOption
 public class ConfigurationClientOptions : HttpPipelineOptions {
 
     public ConfigurationClientOptions(ServiceVersion version = ServiceVersion.V2019_05_09) {
-        if (value == default) throw new ArgumentException(...);
+        if (version == default) 
+            new ArgumentException($"The service version {version} is not supported by this library.");
```
Comment (tg-msft) [inline docs/dotnet/introduction.md]: You forgot a `{ throw`.

Comment (KrzysztofCwalina) [reply docs/dotnet/introduction.md]: BTW, I took the message from storage, but the message makes it sound like the exception should be NotSupportedException. Maybe the message should not talk about "support" bat rather "don't pass bogus values". Though I am not sure we can/want to change it in storage.

**Comment by JonathanGiles** [comment]
Just a general observation - Java is using the same naming of 'service version'. I have recently started to hear a little resistance on that name (from one service team only), with suggestions that 'api version' is the more correct name. For me, the name 'service version' is already baked into a few releases, so I'm not likely to change it now. I just wanted to surface this as an FYI in case it merited further discussion.

**Comment by Petermarcu** [comment]
API version sounds too ambiguous. The Java API is also versioned and it targets a Java API version as well. If we were to adjust the guideline, I would think we would need to communicate that it was the services API version which would make it 'service API version'.

**Comment by brjohnstmsft** [comment]
"Service API version" is a lot closer to the terminology we currently use in our documentation for Azure Cognitive Search than "service version", especially given that the actual parameter is called `api-version`.

**Pull request merged**